src/instruction-search.njk

```diff
@@ -19,25 +30,101 @@ permalink: search/index.html
 <script>
 const input = document.getElementById('search');
 const list = document.getElementById('list');
+const filtersFieldset = document.getElementById('extension-filters-fieldset');
+{# the blank area where we’ll insert checkboxes #}
+const filtersContainer = document.getElementById('extension-filters');
 const items = Array.from(list.querySelectorAll('li'));
-input.addEventListener('input', () => {
+const activeExtensions = new Set();
+
+const extensionMap = new Map();
+for (const li of items) {
+  const slug = li.dataset.extensionSlug;
+  const extension = li.dataset.extension;
+  if (slug && extension && !extensionMap.has(slug)) {
+    extensionMap.set(slug, { extension, slug });
+  }
+}
+
+const extensions = Array.from(extensionMap.values()).sort((a, b) => a.extension.localeCompare(b.extension));
```

Comment:
Because we only seem to use `extensionMap` to construct `extensions`, can we skip the `Map` object altogether? That is, just append stuff to the `extensions` array directly and then sort it.